### PR TITLE
chore: introduce changelog generation

### PR DESCRIPTION
**Pull request by TylorS**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,3 +31,15 @@
...
+## Developing
+
+### Requirements
+
+* Latest stable version of [Node](https://github.com/creationix/nvm)
+* Latest stable version of [Yarn](https://yarnpkg.com)
+
+### For generating changelog (for those with NPM publishing rights)
+
+* [Ruby >= v2.1.0](https://www.ruby-lang.org/en/downloads/)
```
Comment (Frikki) [inline README.md]: `gem install rake` is also required on Debian based systems such as Ubuntu and Linux Mint.

Comment (Frikki) [reply README.md]: export `CHANGELOG_GITHUB_TOKEN`

File: package.json [hunk 2]
```diff
@@ -30,9 +30,10 @@
     "test:unit": "lerna run test:unit",
     "test:lint": "lerna exec -- ../../node_modules/.bin/prettier --write --print-width 100 --tab-width 2 --no-semi --single-quote --trailing-comma es5 --parser typescript src/**/*.ts",
     "build": "node tools/build.js",
+    "changelog": "github_changelog_generator --no-compare-link --bug-labels 'Type:\u0020Bug' --enhancement-labels 'Type:\u0020Feature,Type:\u0020Chore' --include-labels 'Type:\u0020Feature,Type:\u0020Chore,Type:\u0020Bug,Type:\u0020Breaking,Type:\u0020Regression' --user motorcyclets --project motorcycle",
```
Comment (Frikki) [inline package.json]: why these `\u0020` codes?

File: README.md (hunk 1 above)
Comment (TylorS) [reply README.md]: Yeah these would be good to add 👍

File: package.json (hunk 2 above)
Comment (TylorS) [reply package.json]: They are spaces, it doesn't seem to work with _actual_ spaces in their place. It errors trying to read them.

Comment (Frikki) [reply package.json]: I thought so.

File: README.md [hunk 3]
```diff
@@ -31,3 +31,39 @@
...
+* [Github Personal Access Token](https://github.com/settings/tokens)
+  - Save the access token you generated for later usage
+
+In your terminal, run the following:
+
+NOTE: `sudo` may be required for all of the following terminal commands -- based
```
Comment (Frikki) [inline README.md]: Remove:

> -- based on where you installed `Ruby`.

It's just blah blah.

File: README.md [hunk 4]
```diff
@@ -31,3 +31,39 @@
...
+
+```sh
+# If you're on a Debian-based Linux Distribution, install rake
+gem install rake
+
+# Install Github Changelog Generator
+gem install github_changelog_generator
+```
+
+In your `.bash_profile`, `.zshrc` or equivalent shell configuration file add the
```
Comment (Frikki) [inline README.md]: Comma after "configuration file".

File: README.md [hunk 5]
```diff
@@ -31,3 +31,39 @@
...
+```
+
+In your `.bash_profile`, `.zshrc` or equivalent shell configuration file add the 
+following:
+
+```sh
+export CHANGELOG_GITHUB_TOKEN=$GITHUB_PERSONAL_ACCESS_TOKEN_GENERATED_ABOVE
+```
+
+and save. In a new terminal you will now be able to successfully run `yarn changelog`
```
Comment (Frikki) [inline README.md]: Remove "and save." We are not dealing with noob computer users who will generate change logs.

Correct:

> In a new terminal, you should now be able to successfully run `yarn changelog`

File: package.json [hunk 6]
```diff
@@ -30,9 +30,10 @@
...
     "docs": "node tools/docs.js && git add **/README.md && git commit -m 'docs(README): rebuild documentation' && git push",
     "release:pre": "yarn test && yarn build",
-    "release:post": "yarn docs",
+    "release:post": "yarn docs && yarn changelog && git add CHANGELOG.md && git commit -m 'docs(CHANGELOG): rebuild changelog && git push'",
```
Comment (Frikki) [inline package.json]: Why this `docs(CHANGELOG):`? It seems so ... well ... unnecessary.

Comment (TylorS) [reply package.json]: Just following the convention still in the `docs` script. Should we look at removing that convention from both in another issue/PR?

Comment (Frikki) [reply package.json]: Hmm! They need to be removed together. If we follow the one-change policy, then another issue/PR. What do you think?

Comment (TylorS) [reply package.json]: Yeah, another issue and PR.

File: README.md [hunk 7]
```diff
@@ -31,3 +31,38 @@
...
+```
+
+In your `.bash_profile`, `.zshrc` or equivalent shell configuration file, add the 
+following:
+
+```sh
+export CHANGELOG_GITHUB_TOKEN=$GITHUB_PERSONAL_ACCESS_TOKEN_GENERATED_ABOVE
+```
+
+In a new terminal, you will now be able to successfully run `yarn changelog`
```
Comment (Frikki) [inline README.md]: Not "will", but _should_. There's no guarantee.

File: README.md [hunk 8]
```diff
@@ -31,3 +31,38 @@
...
+
+### For generating changelog (for those with NPM publishing rights)
+
+* [Ruby >= v2.1.0](https://www.ruby-lang.org/en/downloads/)
+* [Github Personal Access Token](https://github.com/settings/tokens)
+  - Save the access token you generated for later usage
+
+In your terminal, run the following:
+
+NOTE: `sudo` may be required for all of the following terminal commands.
```
Comment (Frikki) [inline README.md]: Correction:

> **Note:**

That is, `**Note:**`.

File: README.md [hunk 9]
```diff
@@ -31,3 +31,38 @@
...
+
+### For generating changelog (for those with NPM publishing rights)
+
+* [Ruby >= v2.1.0](https://www.ruby-lang.org/en/downloads/)
+* [Github Personal Access Token](https://github.com/settings/tokens)
+  - Save the access token you generated for later usage
+
+In your terminal, run the following:
+
+**NOTE**: `sudo` may be required for all of the following terminal commands.
```
Comment (Frikki) [inline README.md]: Almost ... `**Note:**`

**Pull request merged**